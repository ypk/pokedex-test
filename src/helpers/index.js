import GetYear from "./date-ops";

export {
    GetYear
}