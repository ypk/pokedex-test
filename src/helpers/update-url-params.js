const UpdateUrlQueryParams = (url, key, value) => {
    var targetUrl = new URL(url);
    var searchParams = new URLSearchParams(targetUrl.search);
    searchParams.set(key, value);
    targetUrl.search = searchParams.toString();
    return targetUrl;
}

export default UpdateUrlQueryParams;