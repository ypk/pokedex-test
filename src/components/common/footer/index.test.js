import { render, cleanup } from "@testing-library/react";

import Footer from "./index";
import { GetYear } from "../../../helpers";

afterEach(cleanup)

describe("Testing <Footer />", () => {
    describe("renders without items", () => {
        it("should match snapshot", () => {
            const component = render(<Footer />);
            expect(component.asFragment()).toMatchSnapshot();
        });

        describe("rendering footer functionality", () => {
            it("renders footer", () => {
                const currentYear = GetYear();
                const footerText = `© ${currentYear}, ypk. Everythingisopen-sourced.`;
                const { container } = render(<Footer />);
                expect(container).toBeInTheDocument();
                expect(container.textContent.replace(/ /ig, "")).toBe(footerText)
            });
        });

        describe("testing GetYear <-> replace here if test fails", () => {
            it("returns current year", () => {
                const currentYear = GetYear();
                expect(currentYear).toBe(2021)
            });
        });
    });
});


