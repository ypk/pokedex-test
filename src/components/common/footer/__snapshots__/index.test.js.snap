// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Footer /> renders without items should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="text-gray-600 body-font bg-gray-700"
    data-testid="footer"
  >
    <div
      class="bg-gray-900"
    >
      <div
        class="container mx-auto py-4 px-5 flex flex-wrap flex-row text-white text-sm text-center"
      >
        <p
          class="sm:text-left"
        >
          © 2021,
        </p>
         
        <a
          class="text-gray-300"
          href="https://github.com/ypk"
        >
          ypk
        </a>
        . 
        <p>
          Everything is open-sourced.
        </p>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
